realm/realm-library/src/objectServer/java/io/realm/PermissionManager.java

```diff
@@ -1001,12 +1012,48 @@ protected final boolean checkAndReportInvalidState() {
                 return false;
             }
 
+            // Handle Client Reset if it happened in any of the Realms.
+            // A Client Reset is a fatal error for the PermissionManager, so all current and
+            // future tasks will exit as soon as posible after this event happened and report it
+            // through the error callback. Only action a user can take is to close the
+            // PermissionManager and re-open it again Some data might be lost (like permission
+            // offers not yet processed). This is currently unavoidable.
+            // TODO: Eventually we might be able to recover the permission manager from this event
+            // but it will require some serious task management as we would need to do a full
+            // close, reschedule all tasks, and re-open behind users back. This is out of scope for
+            // now.
+            if (managementErrorHappened && managementError instanceof ClientResetRequiredError) {
+                ClientResetRequiredError cr = (ClientResetRequiredError) managementError;
+                permissionManager.managementRealm.close();
```

Comment:
Object store is actually crashing hard in that case. For now I'll implement a check that only allow the `PermissionManager to be opened from the main thread. Since it requires a looper that is probably what 99% of people would do anyway.

I'll create an issue for tracking it as well though.